frontend/src/app/active-sessions/upload-dialog/upload-dialog.component.ts

```diff
@@ -0,0 +1,224 @@
+            this.treeControl.expand(parentNode);
+            break;
+          }
+        }
+      }
+    return result;
+    }
+
+  checkIfFileExists(parentNode: PathNode, fileName: string): boolean {
+    if (!!parentNode.children){
+      for (var i = 0; i < parentNode.children.length; i++) {
+        if (fileName == parentNode.children[i].name) return true;
+    }
+  }
+  return false;
+  }
+
+  expandToNode(node: PathNode): void{
+    this._expandToNode(this.dataSource.value[0], node);
+  }
+
+  _expandToNode(parentNode: PathNode, node: PathNode): boolean {
+    var result = false;
+    if (node === parentNode) {
+      this.treeControl.expand(parentNode);
+      result = true;
+    } else if (!!parentNode.children){
+      for (var i = 0; i < parentNode.children?.length; i++) {
+        result = this._expandToNode(parentNode.children[i], node);
+        if (result) {
+          this.treeControl.expand(parentNode);
+        }
+      }
+    }
+    return result;
+  }
+
+  removeFile(path: string, filename: string): void{
+    this.removeFileFromSelection(path, filename);
+    this.removeFileFromTree(path, filename);
+  }
+
+  findNode(prefix: string, searchedName: string): [PathNode, number] | null {
+    return this._findNode(this.dataSource.value[0], searchedName, prefix)
+  }
+
+  _findNode(parentNode: PathNode, searchedName: string, prefix: string): [PathNode, number] | null {
+    if (parentNode.children!!){
+      for (var i = 0; i < parentNode.children.length; i++) {
+        const child = parentNode.children[i];
+        if (child.name === searchedName && child.path === prefix ){
+          return [parentNode, i]
+        } else {
+          var result = this._findNode(child, searchedName, prefix)
+          if (!!result) {
+            return result
+          }
+        }
+      }
+    }
+    return null
+  }
+
+  removeFileFromTree(path: string, filename: string): void{
+    const result = this.findNode(path, filename)
+    if (!!result) {
+      result[0].children?.splice(result[1], 1);
+      this.dataSource.next([{ ...this.dataSource.value[0] }]);
+      this.expandToNode(result[0]);
+    }
+  }
+
+  removeFileFromSelection(path: string, filename: string): void {
+    var file, prefix = null
+    for (var i = 0; i < this.files.length; i++) {
+      file = this.files[i][0]
+      prefix = this.files[i][1]
+      if (this.files[i][0].name === filename && this.files[i][1] === path) {
+        break;
+      }
+    }
+    if (!!file && !! prefix) {
+      const index: number = this.files.indexOf([file, prefix]);
+      this.files.splice(index, 1);
+    }
+  }
+
+  onSubmit() {
+    const formData = new FormData();
+    this.files.forEach(([file, prefix]: [File, string]) => {
+      formData.append('files', file, `${prefix}/${file.name}`);
+    });
+    formData.append('id', this.session.id);
+
+    const upload$ = this.loadService
+      .upload(this.session.id, formData)
+      .pipe(finalize(() => this.reset()));
```

Comment:
This does also close the dialog if the upload fails. And as you reload the window in `reset()`, there is no error message visible. 
We should close the window only on success.